lib/web_ui/test/engine/surface/shaders/shader_builder_test.dart

```diff
@@ -0,0 +1,199 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// @dart = 2.6
+import 'package:test/bootstrap/browser.dart';
+import 'package:test/test.dart';
+import 'package:ui/src/engine.dart';
+import 'package:ui/ui.dart' hide window;
+
+void main() {
+  internalBootstrapBrowserTest(() => testMain);
+}
+
+void testMain() {
+  String mat2Sample = 'mat2(1.1, 2.1, 1.2, 2.2)';
+  String mat3Sample = 'mat3(1.1, 2.1, 3.1, // first column (not row!)\n'
+      '1.2, 2.2, 3.2, // second column\n'
+      '1.3, 2.3, 3.3  // third column\n'
+      ')';
+  String mat4Sample = 'mat3(1.1, 2.1, 3.1, 4.1,\n'
+      '1.2, 2.2, 3.2, 4.2,\n'
+      '1.3, 2.3, 3.3, 4.3,\n'
+      '1.4, 2.4, 3.4, 4.4,\n'
+      ')';
+
+  setUpAll(() async {
```

Comment:
I think this won't work on ios-safari due to simulator restrictions.

We can add a skip to this test methods using operating system, example: https://github.com/flutter/engine/blob/bada9fc5f34b0145e7a1a38719a48971f23e7019/lib/web_ui/test/engine/surface/surface_test.dart#L205

Reply:
See https://github.com/flutter/goldens/blob/master/engine/web/draw_vertices_triangle_fan.iOS_Safari.png . webgl1 seems to be working in that case.

Reply:
I see so they fixed the bug on the simulator :) this is good to know let me try to see if I can un-skip some other tests :D